app/vmalert/datasource/vm.go

```diff
@@ -49,28 +51,115 @@ const queryPath = "/api/v1/query?query="
 
 // VMStorage represents vmstorage entity with ability to read and write metrics
 type VMStorage struct {
-	c             *http.Client
-	queryURL      string
-	basicAuthUser string
-	basicAuthPass string
+	c                *http.Client
+	baseURL          string
+	suffix           string
+	basicAuthUser    string
+	basicAuthPass    string
+	defaultAuthToken *auth.Token
+}
+
+func parseURL(s string) (string, string, *auth.Token, error) {
```

Comment:
Can we reuse `httpserver.ParsePath()` here?

Reply:
Thought about using `httpserver.ParsePath()`, but this function will raise error if the tenancy was not set.    

Actually I don't like the way that we have to parse the input URL. I am wondering what if we dump `datasource.url` or delete tenancy parts: 
1. add a global tenancy setting maybe like:
    ```yml
    global:
      tenant:
        accountID: 42
        projectID: 12
    groups:
    - name: ...
    ```
2. some command line flags like `tenant.globalAccountID` and `tenant.globalProjectID`.

What do you think? Will this be unfriendly to users?

PS: prefer the first one.

Reply:
Not sure if I understand. I assume there should two options to configure vmalert:
1. Configure vmalert per tenant. User specifies full `datasource.url` with auth token. That's how many people do now and we should should keep compatibility.
2. Configure multiple tenants for one vmalert. User specifies `datasource.url` to point the `vmselect` address only and omits auth token. Then user specifies tenant account and project IDs per-group or per-file (global config from p.1) and vmalert builds the correct URL for each group.
This looks natural and shouldn't break existing installations.

Reply:
So these two options should both be available for vmalert? First one for capacity and second one for long term using, Right?